test/Microsoft.AspNet.Razor.Test/Framework/ParserTestBase.cs

```diff
@@ -277,15 +277,14 @@ public static void EvaluateParseTree(Block actualRoot, Block expectedRoot)
             // Evaluate the result
             var collector = new ErrorCollector();
 
-            // Link all the nodes
-            expectedRoot.LinkNodes();
-
             if (expectedRoot == null)
             {
                 Assert.Null(actualRoot);
             }
             else
             {
+                // Link all the nodes
+                expectedRoot.LinkNodes();
```

Comment:
- curious.  why did this line need to move?
- add a blank line before the `Assert.NotNull()` because that isn't part of linking the nodes.